master/buildbot/scripts/logwatcher.py

```diff
@@ -60,13 +60,14 @@ class LogWatcher(LineOnlyReceiver):
     TIMEOUT_DELAY = 10.0
     delimiter = unicode2bytes(os.linesep)
 
-    def __init__(self, logfile):
+    def __init__(self, logfile, timeout_delay=None):
         self.logfile = logfile
         self.in_reconfig = False
         self.transport = FakeTransport()
         self.pp = TailProcess()
         self.pp.lw = self
         self.timer = None
+        self.timeout_delay = timeout_delay or self.TIMEOUT_DELAY
```

Comment:
this is not a requirement, however it's a good idea to add new *internal* data members as private (i.e. `_name`) unless it's intended to have them as a part of API (I know that Buildbot is not very good in this particular approach, however it caused a lot of pain during re-factoring).